sklearn/decomposition/tests/test_nmf.py

```diff
@@ -103,30 +107,36 @@ def test_nmf_fit_nn_output():
               5. + np.arange(1, 6)]
     for solver in ('cd', 'mu'):
         for init in (None, 'nndsvd', 'nndsvda', 'nndsvdar', 'random'):
-            model = NMF(n_components=2, solver=solver, init=init,
-                        random_state=0)
-            transf = model.fit_transform(A)
-            assert not((model.components_ < 0).any() or
-                       (transf < 0).any())
+            for regularization in (None, 'both',
+                                   'components', 'transformation'):
```

Comment:
this test could be refactored with a few `parameterize` options for `solver`, `init`, and `regularization`. It'd make it look much nicer.

Reply:
What does `@pytest.mark.parametrize("param", <list>)` really do? Sure, I'll refactor the test!